src/apps/lwaftr/acl.lua

```diff
@@ -0,0 +1,76 @@
+module(...,package.seeall)
+
+local filter = require("lib.pcap.filter")
+
+local INGRESS_FILTER_IPV4 = 1
+local INGRESS_FILTER_IPV6 = 2
+local EGRESS_FILTER_IPV4  = 3
+local EGRESS_FILTER_IPV6  = 4
+
+local function trim(s)
+  return s:gsub("^%s*(.-)%s*$", "%1")
+end
+
+local function columns(content, sep)
+   local function iter(state)
+      local token, pos = state.content:match(state.regex, state.start)
+      if not pos then return nil end
+      state.start = pos
+      return token
+   end
+   sep = sep or ","
+   local regex = ("([^%s]+)()%s"):format(sep, sep)
+   local state = { content = content, regex = regex, start = 0 }
+   return iter, state
+end
+
+local function compile_filter(expr)
+   local function filter_to_str(filter)
+      return table.concat(filter, " and ")
+   end
+   return filter:new(filter_to_str(expr))
+end
+
+local function as_filter_table(filters)
+   return {
+      ingress_filter = {
+         ipv4 = compile_filter(filters[INGRESS_FILTER_IPV4]),
+         ipv6 = compile_filter(filters[INGRESS_FILTER_IPV6]),
+      },
+      egress_filter = {
+         ipv4 = compile_filter(filters[EGRESS_FILTER_IPV4]),
+         ipv6 = compile_filter(filters[EGRESS_FILTER_IPV6]),
+      }
+   }
+end
+
+function compile(filename, args)
```

Comment:
This function needs a comment about the grammar of the file and what the arguments mean.  As it, I can't really tell what's going on.  I also wonder why you feel the need to index filters by named number instead of by name (i.e. foo.egress_fitler_ipv4 vs foo[EGRESS_FILTER_IPV4], but that could make sense to me if there were more comments :)